### PR TITLE
refactor(localdebug): make _checkAndInstall() more common

### PR DESCRIPTION
**Pull request by XiaofuHuang**
1. change the interface of port checker, split getPortsInUse to `getPortsFromProject`  and `getPortsInUse`
2. make `_checkAndInstall` more common, move projectSettings, ports and enabled checker order outof `_checkAndInstall`

E2E TEST: https://github.com/OfficeDev/TeamsFx-UI-Test/blob/71b4e45315fea58a7c43dbc2668bb6c0a3068ce0/src/ui-test/localdebug/localdebug-command-and-response.test.ts#L12

### CODE REVIEW

File: packages/vscode-extension/src/debug/prerequisitesHandler.ts [hunk 1]
```diff
@@ -263,55 +291,62 @@ async function checkPort(
...
+// export async function checkAndInstallV2(
+//   orderedCheckers: PrerequisiteOrderedChecker[],
+//   ports: number[]
+// ): Promise<Result<void, FxError>> {
+// }
+
+async function _checkAndInstall(
+  logLabel: string,
+  orderedCheckers: PrerequisiteOrderedChecker[],
+  ports: number[]
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/prerequisitesHandler.ts:348]: It's weird to have `ports` as input. `ports` is bound with port checker, then what if `orderedCheckers` does not contain port checker?
Is it possible to move `ports` into `checkPort`?

Comment (XiaofuHuang) [reply packages/vscode-extension/src/debug/prerequisitesHandler.ts:348]: I try to move all the configuration related to `projectSettings` out of checker. What about move the args (ports) into PrerequisiteOrderedChecker?

File: packages/vscode-extension/src/debug/prerequisitesHandler.ts [hunk 2]
```diff
@@ -131,6 +142,27 @@ const ProgressMessage: { [key: string]: string } = Object.freeze({
   [DepsType.FuncCoreTools]: `Checking and installing ${DepsDisplayName[DepsType.FuncCoreTools]}`,
 });
 
+type PortCheckerInfo = { checker: Checker.Ports; ports: number[] };
+type PrerequisiteCheckerInfo = { checker: Checker | DepsType } | PortCheckerInfo;
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/prerequisitesHandler.ts:146]: `PrerequisiteCheckerInfo` can be cleaner without `| PortCheckerInfo`.
Just convert to `PortCheckerInfo` for port checker. `(info as PortCheckerInfo).ports...`

**Review by swatDong — DISMISSED** [review]

**Comment by XiaofuHuang** [comment]
E2E TEST: https://github.com/OfficeDev/TeamsFx-UI-Test/blob/71b4e45315fea58a7c43dbc2668bb6c0a3068ce0/src/ui-test/localdebug/localdebug-command-and-response.test.ts#L12

**Review by swatDong — APPROVED** [review]

**Pull request merged**